e2e/awspcaissuer_test.go

```diff
@@ -124,22 +127,22 @@ func InitializeTestSuite(suiteCtx *godog.TestSuiteContext) {
 		}
 
 		// Create CAs to be used in testing
-		testContext.caArns["RSA"] = createCertificateAuthority(ctx, cfg, true)
+		testContext.createCertificateAuthority(ctx, cfg, "RSA", types.SigningAlgorithmSha256withrsa)
 		log.Printf("Created RSA CA with arn %s", testContext.caArns["RSA"])
 
-		testContext.caArns["ECDSA"] = createCertificateAuthority(ctx, cfg, false)
+		testContext.createCertificateAuthority(ctx, cfg, "ECDSA", types.SigningAlgorithmSha256withecdsa)
 		log.Printf("Created EC CA with arn %s", testContext.caArns["ECDSA"])
 
 		xaRole, xaRoleExists := os.LookupEnv(CrossAccountRoleKey)
 		if xaRoleExists {
 			testContext.xaCfg = assumeRole(ctx, cfg, xaRole, testContext.region)
 
-			testContext.caArns["XA"] = createCertificateAuthority(ctx, testContext.xaCfg, true)
+			testContext.createCertificateAuthority(ctx, testContext.xaCfg, "XA", types.SigningAlgorithmSha256withrsa)
 
 			log.Printf("Created XA CA with arn %s", testContext.caArns["XA"])
 
-			endEntityResourcePermission := "arn:aws:ram::aws:permission/AWSRAMDefaultPermissionCertificateAuthority"
-			subordinateCaResourcePermission := "arn:aws:ram::aws:permission/AWSRAMSubordinateCACertificatePathLen0IssuanceCertificateAuthority"
+			endEntityResourcePermission := "arn:" + testContext.partition + ":ram::aws:permission/AWSRAMDefaultPermissionCertificateAuthority"
+			subordinateCaResourcePermission := "arn:" + testContext.partition + ":ram::aws:permission/AWSRAMSubordinateCACertificatePathLen0IssuanceCertificateAuthority"
```

Comment:
nit: you could also do sprintf... but I think this is fine. I am not sure which is more preferable in Golang.